src/geeknn/ordination/gnn.py

```diff
@@ -1,218 +1,38 @@
-import ee
+from __future__ import annotations
 
-from . import utils
-from .cca import cca
+import numpy as np
+from sknnr.transformers import CCATransformer
 
-# TODO: Maybe make dataclass if GEE can handle that
-# TODO: Don't assign self variables in train
+from ._base import Transformed
 
 
-class GNN:
+class GNN(Transformed):
+    SPP_TRANSFORM_FUNC = {
+        "SQRT": lambda x: np.sqrt(x),
+        "LOG": lambda x: np.log(x),
+        "NONE": lambda x: x,
```

Comment:
I think I'm being dense here, but I'm trying to figure out how to make this change, but are you suggesting something like this?
```
class GNN(Transformed):
    Y_TRANSFORM_FUNC = {
        "SQRT": lambda x: np.sqrt(x),
        "LOG": lambda x: np.log(x),
    }

    ...

    def _get_modeling_X_y(self, *, client_fc, X_columns, y_columns):
        X = client_fc.properties_to_array(X_columns)
        y = client_fc.properties_to_array(y_columns)
        transform_func = self.Y_TRANSFORM_FUNC.get(self.y_transform.upper(), lambda x: x)
        y = transform_func(y)
```
I don't think this is what you meant exactly ...

Reply:
I think what I had in mind (but didn't explain well) was using `None` as the key instead of the string `"NONE"`, so the user could specify `y_transform=None` to apply the no-op transform (I had to check to confirm, but `None` is a valid dict key). Returning the no-op as a default from `get` is a little more concise though. My main goal was just to remove a string option since I find them hard to remember and a little messy to validate.

Reply:
Now I understand, thanks for clarifying.  I'm going with your advice to make this directly a callable.  Much cleaner and no validation on the strings as you say.